packages/big-design/src/components/TableNext/SelectAll/useSelectAllState.ts

```diff
@@ -0,0 +1,68 @@
+import { TableExpandable, TableSelectable } from '../types';
+
+import {
+  areAllInPageSelected,
+  areSomeInPageSelected,
+  getTotalSelectedItems,
+  selectAll,
+  unselectAll,
+} from './helpers';
+
+interface useSelectAllStateProps<T> {
+  isExpandable: boolean;
+  items: T[];
+  selectedItems: TableSelectable['selectedItems'];
+  expandedRowSelector?: TableExpandable<T>['expandedRowSelector'];
+  onChange?: TableSelectable['onSelectionChange'];
+}
+
+export const useSelectAllState = <T>({
+  expandedRowSelector,
+  isExpandable,
+  items,
+  selectedItems,
+  onChange,
+}: useSelectAllStateProps<T>) => {
+  const allInPageSelected = areAllInPageSelected(
+    isExpandable,
+    items,
+    selectedItems,
+    expandedRowSelector,
+  );
+
+  const someInPageSelected = areSomeInPageSelected(
+    isExpandable,
+    items,
+    selectedItems,
+    expandedRowSelector,
+  );
+
+  const totalSelectedItems = getTotalSelectedItems(items, selectedItems);
+  const label = allInPageSelected ? 'Deselect All' : 'Select All';
+
+  const handleSelectAll = () => {
+    if (typeof onChange !== 'function') {
+      return;
+    }
+
+    const newSelectedItems = { ...selectedItems };
+
+    if (allInPageSelected) {
+      selectAll(isExpandable, items, newSelectedItems, expandedRowSelector);
+
+      return onChange(newSelectedItems);
+    }
+
+    unselectAll(isExpandable, items, newSelectedItems, expandedRowSelector);
+
+    return onChange(newSelectedItems);
```

Comment:
Why does it look like the `selectAll` and `unselectAll` are reversed 😆 . I'm sure it's working fine, just having hard time in understanding why we wouldn't `unselectAll` if `allInPageSelected`?

Reply:
lol, good catch, thanks! 😓  I swapped the names